File: 백준/Silver/10866. 덱/덱.java
import java.io.*;
import java.util.Deque;
import java.util.LinkedList;

public class Main {

    public static void main(String[] args) throws IOException {
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int N = Integer.parseInt(br.readLine());
        Deque<Integer> dq = new LinkedList<>();
        String[] inputs;
        for (int i = 0; i < N; i++) {
            inputs = br.readLine().split(" ");
            String op = inputs[0];

            if (op.equals("size")) {
                bw.write(dq.size() + "\n");
            } else if (op.equals("empty")) {
                bw.write(dq.isEmpty() ? "1\n" : "0\n");
            } else if (op.equals("front")) {
                bw.write(dq.isEmpty() ? "-1\n" : dq.peekFirst()+"\n");
            } else if (op.equals("back")) {
                bw.write(dq.isEmpty() ? "-1\n" : dq.peekLast()+"\n");
            } else if (op.equals("pop_front")) {
                bw.write(dq.isEmpty() ? "-1\n" : dq.pollFirst()+"\n");
            } else if (op.equals("pop_back")) {
                bw.write(dq.isEmpty() ? "-1\n" : dq.pollLast()+"\n");
            } else if (op.equals("push_back")) {
                dq.add(Integer.parseInt(inputs[1]));
            } else if (op.equals("push_front")) {
                dq.addFirst(Integer.parseInt(inputs[1]));
            }
        }
        bw.flush();
    }
}

File: 백준/Silver/10866. 덱/덱.py
from collections import deque
import sys
if __name__ == "__main__":

    N = int(sys.stdin.readline())
    dq = deque()

    for _ in range(N):
        inputs = sys.stdin.readline().split()
        op = inputs[0]
        if op == 'size':
            print(len(dq))

        elif op == 'empty':
            if len(dq) == 0:
                print(1)
            else:
                print(0)

        elif op == 'front':
            if len(dq) == 0:
                print(-1)
            else:
                print(dq[0])

        elif op == 'pop_front':
            if len(dq) == 0:
                print(-1)
            else:
                print(dq.popleft())

        elif op == 'back':
            if len(dq) == 0:
                print(-1)
            else:
                print(dq[-1])
        elif op == 'pop_back':
            if len(dq) == 0:
                print(-1)
            else:
                print(dq.pop())
        elif op == 'push_front':
            dq.appendleft(int(inputs[1]))
        elif op == 'push_back':
            dq.append(int(inputs[1]))
